src/MongoDB/Driver/Cursor.cpp

```diff
@@ -323,6 +315,9 @@ void HHVM_METHOD(MongoDBDriverCursor, setTypeMap, const Array &typemap)
 	MongoDBDriverCursorData* data = Native::data<MongoDBDriverCursorData>(this_);
 
 	parseTypeMap(&data->bson_options, typemap);
+
+	/* Reconvert the current document to use the new typemap */
+	hippo_cursor_load_current(data);
```

Comment:
HHVM pre-populates the first item in the result. This is before you can call ->setTypemap(). If it doesn't re-encode with the new typemap, it had used the default one, and that's *not* what you want to expect as a result. There is no test needed, as without it, all tests that set a typemap fail.

Before the change to the new API I could work around this, but no more. I have created a ticket to see whether we can do something about this in the future: https://jira.mongodb.org/browse/HHVM-218